### ISSUE TITLE
Error card.dueDate and lastReviewed are null

### ISSUE THREAD

**Issue by Mozoku** [JobiJoba/dolphhinsr_dart#6]
The reviews and masters load well from the sembast db. For some reason, the cards are still null. Thanks

**Comment by JobiJoba** [comment]
Hey, 

So, you add the master first, then add the reviews, but dueDate and lastReviewed are null right ? 

Can I see some code ?

**Comment by Mozoku** [comment]
Load masters from db
<img width="515" alt="Screenshot 2020-12-08 at 11 02 27" src="https://user-images.githubusercontent.com/26275146/101456390-07200e00-3945-11eb-803d-efee8517a4af.png">
user triggered action to update reviews
<img width="498" alt="Screenshot 2020-12-08 at 11 03 06" src="https://user-images.githubusercontent.com/26275146/101456404-0ab39500-3945-11eb-9403-4096f0e824c2.png">

**Comment by Mozoku** [comment]
> Hey,
> 
> So, you add the master first, then add the reviews, but dueDate and lastReviewed are null right ?
> 
> Can I see some code ?

Yes. That's what happens. I'd also like to add some features like decks to the package.

**Comment by JobiJoba** [comment]
Hey Mozoku, 
In your didDependencies change after your dolphin.AddMasters(prov.masterList); 
you also have to do something like : 

dolphin.addReviews(prov.reviewList) with the previous reviews of the user (so you have to save the reviews in your DB also).

About the Decks my app also have many Decks and what I did was that I create an instance of DolphinSR for each Decks and put it in a map like this: 


```
Map<String, DolphinSR> dolphins;

... 

dolphins["MyDeck"] = DolphinSR(); 
... 


dolphins["MyDeck"].addMasters(prov.masterOfMyDeck);
dolphins["MyDeck"].addReviews(prov.reviewsOfMyDeckForUser);

```


Also, just to be sure you are on the version 2.0.2  right ? :)

**Comment by Mozoku** [comment]
Yeah, I'm on 2.0.2. I do save the reviews in my db. The reviews are loaded successfully from the db. 

> "Unhandled Exception: NoSuchMethodError: The getter 'mode' was called on null.
>  Receiver: null
> Tried calling: mode"

This occurs when I try to call next card. What could the issue be?

**Comment by JobiJoba** [comment]
Mmmm I think the errors happens here when you apply the old reviews (maybe an issue of IDs or something)? 
(seems that state is null) 

```
String computeScheduleFromCardState(CardState state, DateTime now) {
  if (state.mode == 'lapsed' || state.mode == 'learning') {
    return 'learning';
  } else if (state.mode == 'reviewing') {
    final diff = dateDiffInDays(calculateDueDate(state), now);
    if (diff < 0) {
      return 'later';
    } else if (diff >= 0 && diff < 1) {
      return 'due';
    } else if (diff >= 1) {
      return 'overdue';
    }
  }

  throw Exception('Issue with mode and calculation of a cardState');
}
```
What  you can do is download the package on your computer and link it like this in  your pubspec.yaml

```
 dolphinsr_dart:
    path: /Users/pathToYourProject/Projects/DartModule/dolphhinsr_dart
```

and try to see where it crash in debug. 

I've translated the module a long time ago from Javascript and I must say I don't fully understand what it does but seems to work for me until now. (I would love to fully rewrite it :P)
